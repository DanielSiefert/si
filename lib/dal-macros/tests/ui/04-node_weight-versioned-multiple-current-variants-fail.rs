pub struct InnerType;

pub trait SiVersionedNodeWeight {
    type Inner;

    fn inner(&self) -> &Self::Inner;
    fn inner_mut(&mut self) -> &mut Self::Inner;
}

#[derive(dal_macros::SiVersionedNodeWeight)]
pub enum VersionedNodeWeight {
    #[si_versioned_node_weight(current)]
    V1(InnerType),
    #[si_versioned_node_weight(current)]
    V2(InnerType),
    V3(InnerType),
    V4(InnerType),
    #[si_versioned_node_weight(current)]
    V5(InnerType),
}

fn main() {}
