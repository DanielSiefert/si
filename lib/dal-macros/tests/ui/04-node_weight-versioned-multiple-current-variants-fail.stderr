error: Enum cannot have multiple current variants.
  --> tests/ui/04-node_weight-versioned-multiple-current-variants-fail.rs:14:5
   |
14 | /     #[si_versioned_node_weight(current)]
15 | |     V2(InnerType),
   | |_________________^

error: Enum cannot have multiple current variants.
  --> tests/ui/04-node_weight-versioned-multiple-current-variants-fail.rs:18:5
   |
18 | /     #[si_versioned_node_weight(current)]
19 | |     V5(InnerType),
   | |_________________^
